bundles/framework/mapfull/instance.js

```diff
@@ -470,37 +470,20 @@ Oskari.clazz.define('Oskari.mapframework.bundle.mapfull.MapFullBundleInstance',
                 }
             }
 
-            /* Change to this once plugins can handle it...
-            var plugins = mapmodule.getPluginInstances(),
-                plugin,
-                pluginName;
+            var plugins = mapmodule.getPluginInstances();
+            var selectedLayers = sandbox.findAllSelectedMapLayers();
+            var plugin;
+            var pluginName;
+            var selectedLayer;
 
             for (pluginName in plugins) {
                 plugin = plugins[pluginName];
                 if (plugin && plugin.setState) {
                     plugin.setState(state.plugins[pluginName]);
                 }
-            } */
-
-            // Hackhack
-            if (!state.plugins) {
-                state.plugins = {};
-            }
-
-            if (!state.plugins.MainMapModuleMarkersPlugin) {
-                state.plugins.MainMapModuleMarkersPlugin = {};
-            }
-
-            var plugins = mapmodule.getPluginInstances(),
-                plugin,
-                pluginName;
-
-            for (pluginName in state.plugins) {
-                if (state.plugins.hasOwnProperty(pluginName)) {
-                    // Not finding the plugin is not that uncommon, just move on
-                    plugin = plugins[pluginName];
-                    if (plugin && plugin.setState) {
-                        plugin.setState(state.plugins[pluginName]);
+                if (plugin && plugin.updateLayerStyle) {
+                    for (selectedLayer in selectedLayers) {
+                        plugin.updateLayerStyle(selectedLayers[selectedLayer]);
```

Comment:
Is this needed if the style for layer is set BEFORE AddMapLayerRequest is sent a few lines before?
https://github.com/oskariorg/oskari-frontend/blob/5b3fed8d2f640721a34ede490fa6cb8784c0244d/bundles/framework/mapfull/instance.js#L468

https://github.com/oskariorg/oskari-frontend/blob/5b3fed8d2f640721a34ede490fa6cb8784c0244d/bundles/framework/mapfull/instance.js#L450-L453